readthedocs/projects/forms.py

```diff
@@ -175,6 +175,24 @@ class Meta(object):
         widget=forms.Textarea,
     )
 
+    def clean_language(self):
+        language = self.cleaned_data['language']
+        project = self.instance
+        if project:
+            if project.translations.filter(language=language).exists():
+                msg = 'There is a translation for the "{lang}" language.'
+                raise forms.ValidationError(
+                    _(msg).format(lang=language)
+                )
+            main_project = project.main_language_project
+            if main_project and main_project.language == language:
+                msg = (
+                    'The translation can not have the same '
+                    'language as the parent.'
```

Comment:
So two things:

* We should also be testing for overlapping languages in sibling translations. That is, we should be checking for `project.main_language_project.translations`
* I think we can reduce all the error messages to something more clear: `There is already a {lang} translation for the {proj} project`